src/main/kotlin/eli/inevitable/formatters/TableFormatter.kt

```diff
@@ -0,0 +1,268 @@
+                footerHorizontalDividerToken = formatter.headerHorizontalDividerToken
+                footer = formatter.footer
+            }
+        }
+
+        constructor(): super(TableFormatter())
+
+        constructor(vararg contents: RowFormatter): this(contents.toList())
+
+        constructor(contents: List<RowFormatter>): this() {
+            buildable.bodyContents.addAll(contents)
+        }
+
+        companion object {
+            @JvmStatic
+            fun of(original: TableFormatter): Builder = Builder(original)
+
+            @JvmStatic
+            fun of(): Builder = Builder()
+
+            @JvmStatic
+            fun of(vararg contents: RowFormatter): Builder = Builder(contents.toList())
+
+            @JvmStatic
+            fun of(contents: List<RowFormatter>): Builder = Builder(contents)
+        }
+
+        /**
+         * Resets the contents and dimensions of the [table][TableFormatter] but retains formatting options.
+         * @return This [Builder]
+         */
+        fun resetContents(): Builder {
+            checkFinalizedStatus()
+            buildable.run {
+                lines.clear()
+                bodyContents.clear()
+                columnWidths.clear()
+                width = 0
+                height = 0
+            }
+
+            return this
+        }
+
+        /**
+         * @param contents The [array of rows][RowFormatter] to populate the rows of the table. This will append to any prior rows already specified.
+         *
+         * All rows must have the same number of columns as any other row in the table.
+         * @return This [Builder]
+         */
+        fun body(vararg contents: RowFormatter): Builder {
+            return body(contents.toList())
+        }
+
+        /**
+         * @param contents The [list of rows][RowFormatter] to populate the rows of the table. This will append to any prior rows already specified.
+         *
+         * All rows must have the same number of columns as any other row in the table.
+         * @return This [Builder]
+         */
+        fun body(contents: List<RowFormatter>): Builder {
+            checkFinalizedStatus()
+            buildable.bodyContents.addAll(contents)
+
+            return this
+        }
+
+        /**
+         * @param row The [header row][RowFormatter] for the table.
+         *
+         * Must have the same number of columns as any other row in the table.
+         * @return This [Builder]
+         */
+        fun header(row: RowFormatter): Builder {
+            checkFinalizedStatus()
+            buildable.header = row
+
+            return this
+        }
+
+        /**
+         * @param verticalToken The [vertical token][Char] for the header divider.
+         * @param horizontalToken The [horizontal token][Char] for the header divider.
+         *
+         * @return This [Builder]
+         */
+        fun headerDivider(verticalToken: Char, horizontalToken: Char): Builder {
```

Comment:
since they're both chars, did you consider using a Divider object as an argument?

```
headerDivider(Divider.().vertical('|').horizontal('-'))
```

I think that makes it clearer which is which, but for some reason I think I like the 2 arg method instead :(

Reply:
I figured you were going to comment on that point. Yes, I have considered
passing a divider directly. That was actually part of the original
implementation. However, I ran into the problem of having to construct a
divider with bogus data for each of the columns only to have the table do
the actual resizing later. So that was an option, but seemed inelegant,
because you have to force it into a valid bogus state in order to build the
first divider, only to reconstruct it to make it fit the columns.

The other option was to have it consume a DivFormatter.Builder and have the
table modify it before finalizing. That also seemed less than ideal.

So I landed on just passing in the divider tokens if one is desired. This
way also seems clunky, but less so than the other approaches.

Are you suggesting that I create another Divider object within the context of the Table and Block builders, that wraps around the DivFormatter to build it? I considered that as well, but it also seems like an unnecessary abstraction for the level of complexity of the Divider. Your syntax for that Divider object just seems like it would make it harder to consume and understand. But myabe I'm wrong on that...